clients/drcachesim/tests/scheduler_launcher.cpp

```diff
@@ -0,0 +1,201 @@
+                           .get_input_stream_interface(stream->get_input_stream_ordinal())
+                           ->get_instruction_ordinal()
+                    << " instrs == thread " << record.instr.tid << "\n";
+                std::cerr << line.str();
+            }
+            prev_tid = record.instr.tid;
+        }
+    }
+}
+
+} // namespace
+
+int
+_tmain(int argc, const TCHAR *targv[])
+{
+    // Convert to UTF-8 if necessary
+    char **argv;
+    drfront_status_t sc = drfront_convert_args(targv, &argv, argc);
+    if (sc != DRFRONT_SUCCESS)
+        FATAL_ERROR("Failed to process args: %d", sc);
+
+    std::string parse_err;
+    if (!droption_parser_t::parse_argv(DROPTION_SCOPE_FRONTEND, argc, (const char **)argv,
+                                       &parse_err, NULL) ||
+        op_trace_dir.get_value().empty()) {
+        FATAL_ERROR("Usage error: %s\nUsage:\n%s", parse_err.c_str(),
+                    droption_parser_t::usage_short(DROPTION_SCOPE_ALL).c_str());
+    }
+
+    scheduler_t scheduler;
+    std::vector<scheduler_t::input_workload_t> sched_inputs;
+    sched_inputs.emplace_back(op_trace_dir.get_value());
+    scheduler_t::scheduler_options_t sched_ops(
+        scheduler_t::MAP_TO_ANY_OUTPUT, scheduler_t::DEPENDENCY_IGNORE,
+        scheduler_t::SCHEDULER_DEFAULTS, op_verbose.get_value());
+    sched_ops.quantum_duration = op_sched_quantum.get_value();
+#ifdef HAS_ZLIB
+    std::unique_ptr<zipfile_ostream_t> record_zip;
+    std::unique_ptr<zipfile_istream_t> replay_zip;
+    if (!op_record_file.get_value().empty()) {
+        record_zip.reset(new zipfile_ostream_t(op_record_file.get_value()));
+        sched_ops.schedule_record_ostream = record_zip.get();
+    }
+    if (!op_replay_file.get_value().empty()) {
+        replay_zip.reset(new zipfile_istream_t(op_replay_file.get_value()));
+        sched_ops.schedule_replay_istream = replay_zip.get();
+        sched_ops.mapping = scheduler_t::MAP_AS_PREVIOUSLY;
+    }
+#endif
+    if (scheduler.init(sched_inputs, op_num_cores.get_value(), sched_ops) !=
+        scheduler_t::STATUS_SUCCESS) {
+        FATAL_ERROR("failed to initialize scheduler: %s",
+                    scheduler.get_error_string().c_str());
+    }
+
+    std::vector<std::thread> threads;
+    std::vector<std::vector<memref_tid_t>> schedules(op_num_cores.get_value());
+    std::cerr << "Creating " << op_num_cores.get_value() << " simulator threads\n";
+    threads.reserve(op_num_cores.get_value());
+    for (int i = 0; i < op_num_cores.get_value(); ++i) {
+        threads.emplace_back(std::thread(&simulate_core, i, scheduler.get_stream(i),
+                                         std::ref(scheduler), std::ref(schedules[i])));
+    }
+    for (std::thread &thread : threads)
+        thread.join();
+
+    for (int i = 0; i < op_num_cores.get_value(); ++i) {
+        std::cerr << "Core #" << i << ": ";
+        for (memref_tid_t tid : schedules[i])
+            std::cerr << tid << " ";
```

Comment:
Count of instrs executed by the tid on this core isn't useful?

Reply:
I made this off by default when I was trying to make a test out of this and was having trouble with non-atomic long writes and with ordinals not matching; but since the scheduler_unit_tests now contains that test, yes we can make this print by default.  Done.